### DISCUSSION TITLE
Code error 400 by API response

### DISCUSSION THREAD

**Discussion by Enferum** [pocketbase/pocketbase#709] [Q&A]
Hi, I created a few collections and changed API rules with access for anyone
![2022-10-04_19-02](https://user-images.githubusercontent.com/19825000/193869024-e60fb55e-2f9e-4e59-869f-0693684e39f0.png)

I use Postman for a testing response from the backend
![2022-10-04_19-07](https://user-images.githubusercontent.com/19825000/193869882-9b2174d2-3bdd-431c-8bdf-4855c6938032.png)

I checked headers but didn't find any issues. I changed API rules for only admin \ login as admin \ sent request for getting Product List but the response was the same 400 Failed to load the submitted data due to invalid formatting.

____________________________________________

And I hit the problem with API rules related to user role
create a new multiple choice field with roles
![2022-10-04_19-14](https://user-images.githubusercontent.com/19825000/193871375-ab1571a2-4ff2-42ac-a109-1fdda6bc6f88.png)

Add a new API rule for Create method
![2022-10-04_19-15](https://user-images.githubusercontent.com/19825000/193871800-9ad6aac1-7ca1-4534-87d1-ad5b43cc2693.png)

Logged as a user with role manager
![2022-10-04_19-27](https://user-images.githubusercontent.com/19825000/193874206-dd416a44-9595-4bc3-a1f9-f6ea1bb60981.png)

Prepare auth settings\headers\body but got "code": 400, "message": "Failed to create record.",
![2022-10-04_19-27_1](https://user-images.githubusercontent.com/19825000/193874528-a5cbeebc-6be3-42ff-aaba-a4846363182d.png)

![2022-10-04_19-25](https://user-images.githubusercontent.com/19825000/193874606-55e27a92-9703-475b-81cd-8a3cef5e57cf.png)

![2022-10-04_19-25_1](https://user-images.githubusercontent.com/19825000/193874654-df01eb20-f9f8-45ff-bf00-c0971dae1872.png)

Thanks for any suggestion how to fix it

**Comment by ganigeorgiev** [comment]
The submitted header doesn't seem correct. The last screenshot shows that you are using form-data (aka multipart) while sending `Content-Type: application/json`.

Try unchecking/removing all headers and send the request again with the form-data filled (for multipart/form-data you don't need to specify a header since it is calculated dynamically; for json - you have to use the "raw" radio option in postman).

_I'll move the issue to Q&A discussion._

**Reply by Enferum** [reply]
You're right! I change `Content-Type: application/json` to `multipart/form-data; boundary=<calculated when request is sent>` and rules for roles works well.
Mother of God... I'm so sloppy... for the first case, I didn't change POST to GET method. Sorry for the stupid question.
Thanks a lot for the rapid answer.
